File: Data Structures & Algorithms/two-integer-sum/submission-3.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> sumMap = new HashMap<Integer, Integer>();
        // int[] answers = new int[2];
        for (int index = 0; index < nums.length; index++) {
            if (sumMap.containsKey(target-nums[index])) {
                // answers[1] = index;
                // answers[0] = sumMap.get(target-nums[index]);
                return new int[] {sumMap.get(target-nums[index]), index};
                // break;
            }
            sumMap.put(nums[index], index);
        }

        return new int[0];

    }
}

File: Data Structures & Algorithms/binary-search/submission-3.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        start = 0
        end = len(nums)
        
        while start < end:
            mid = start + (end - start) // 2
            print(mid)
            if (nums[mid] > target):
                end = mid - 1
            elif (nums[mid] < target):
                start = mid + 1
            else:
                return mid
        if start < len(nums) and nums[start] == target:
            return start
        elif end < len(nums) and nums[end] == target:
            return end
        return -1
